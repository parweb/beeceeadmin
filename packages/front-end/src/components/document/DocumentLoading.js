import { Spinner } from '@salesforce/design-system-react';

const DocumentLoading = () => (
  <Spinner
    size="small"
    variant="base"
    assistiveText={{ label: 'Recherche en cours' }}
  />
);

export default DocumentLoading;
