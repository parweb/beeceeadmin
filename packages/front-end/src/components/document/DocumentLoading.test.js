import renderer from 'react-test-renderer';
import DocumentLoading from './DocumentLoading';
import { shallow } from 'enzyme';

describe('<DocumentLoading />', () => {
  test('render correctly DocumentLoading component', () => {
    const component = renderer.create(<DocumentLoading />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Check if spinner exists', () => {
    const wrapper = shallow(<DocumentLoading />);
    expect(wrapper.find('SLDSSpinner')).toHaveLength(1);
  });
});
