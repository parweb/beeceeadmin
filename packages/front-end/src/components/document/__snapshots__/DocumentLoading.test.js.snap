// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocumentLoading /> render correctly DocumentLoading component 1`] = `
<div
  className="slds-spinner_container"
>
  <div
    aria-hidden="false"
    className="slds-spinner slds-spinner_small"
    role="status"
  >
    <span
      className="slds-assistive-text"
    >
      Recherche en cours
    </span>
    <div
      className="slds-spinner__dot-a"
    />
    <div
      className="slds-spinner__dot-b"
    />
  </div>
</div>
`;
